assets/js/index.js

```diff
@@ -0,0 +1,847 @@
+
+		var htmlStylesObject = {};
+
+		for (var attribute in htmlStyles) {
+			htmlStylesObject[attribute] = htmlStyles[attribute];
+		}
+
+		twentytwenty.prevlockStyles = htmlStylesObject;
+	},
+
+	// Lock the scroll (do not call this directly)
+	lock: function () {
+		var appliedLock = {};
+
+		if (twentytwenty.scrollLocked) {
+			return;
+		}
+
+		// Save scroll state and styles
+		twentytwenty.prevScroll = {
+			scrollX: window.scrollX,
+			scrollY: window.scrollY,
+		};
+
+		this.saveStyles();
+
+		// Compose our applied CSS, with scroll state as styles
+		Object.assign(appliedLock, twentytwenty.lockStyles, {
+			left: -window.scrollX + 'px',
+			top: -window.scrollY + 'px',
+		}
+		);
+
+		// Then lock styles and state
+		var html = document.querySelector('html');
+
+		Object.keys(appliedLock).forEach(function (style) {
+			html.style[style] = appliedLock[style];
+		});
+
+		window.scrollX = 0;
+		window.scrollY = 0;
+
+		twentytwenty.scrollLocked = true;
+	},
+
+	// Unlock the scroll (do not call this directly)
+	unlock: function () {
+		if (!twentytwenty.scrollLocked) {
+			return;
+		}
+
+		var html = document.querySelector('html');
+		// Revert styles and state
+		Object.keys(twentytwenty.prevlockStyles).forEach(function (style) {
+			html.style[style] = twentytwenty.prevlockStyles[style];
+		});
+
+		window.screenX = twentytwenty.prevScroll.scrollX;
+		window.scrollY = twentytwenty.prevScroll.scrollY;
+
+		twentytwenty.scrollLocked = false;
+	},
+
+	// Call this to lock or unlock the scroll
+	setTo: function (on) {
+		// If an argument is passed, lock or unlock accordingly
+		if (arguments.length) {
+			if (on) {
+				this.lock();
+			} else {
+				this.unlock();
+			}
+			// If not, toggle to the inverse state
+		} else if (twentytwenty.scrollLocked) {
+			this.unlock();
+		} else {
+			this.lock();
+		}
+	},
+
+}; // twentytwenty.scrollLock
+
+/*	-----------------------------------------------------------------------------------------------
+	Toggles
+--------------------------------------------------------------------------------------------------- */
+
+twentytwenty.toggles = {
+
+	init: function () {
+		// Do the toggle
+		this.toggle();
+
+		// Check for toggle/untoggle on resize
+		this.resizeCheck();
+
+		// Check for untoggle on escape key press
+		this.untoggleOnEscapeKeyPress();
+	},
+
+	// Do the toggle
+	toggle: function () {
+		document.querySelectorAll('*[data-toggle-target]').forEach(function (element) {
+			element.addEventListener('click', function () {
+				// Get our targets
+				var toggle = element;
+				var targetString = toggle.dataset.toggleTarget;
+				var target;
+
+				if (targetString === 'next') {
+					target = toggle.nextSibling;
+				} else {
+					target = document.querySelector(targetString);
+				}
+
+				// Trigger events on the toggle targets before they are toggled
+				if (target.classList.contains('active')) {
+					target.dispatchEvent(new Event('toggle-target-before-active'));
```

Comment:
Seems like `Event()` isn't supported by IE

https://developer.mozilla.org/en-US/docs/Web/API/Event/Event

😞

Reply:
I've tested this [polyfill ](https://github.com/Financial-Times/polyfill-library/blob/master/polyfills/Event/polyfill.js) and the menu and search modal work as intended...

Reply:
I did not see the problem with the event coming. I will look into it and see wether there is a polyfill for it or wether we can rework the code to kit need the dispatching of custom events

Reply:
I think I found a solution for it. I sadly do not have access to a machine running ie11 so my testing is limited. If you have the time I would love to get some feedback wether it is working now :)